### ISSUE TITLE
Solution sanity check needed

### ISSUE THREAD

**Issue by regas** [IdentityServer/IdentityServer3#2603]
Hey,

Great work on IdentityServer, like it very much so far. 
Need an opinion if I am doing my solution the right way and some suggestions.
We have a 3rd party native app client, IdSrv and native app for authentication. Authorization code flow is used. Everything happens inside mobile device.

- Client opens browser and calls IdSrv to authorize
- IdSrv uses external identity provider in native app, which means custom OWIN authentication middleware get challenge to authenticate, redirects browser to another page where a custom URI scheme is fed to browser and native application is started for user to authenticate.
- User authenticates, gives consent to share his data.
- Native app opens another browser window with redirect to IdSrv to send user's identity (not sure how to do this right now at all).
- custom OWIN authentication middleware gets the request, validates, extracts claims, creates ClaimsIdentity and issues AuthenticationTicket and then redirects to IdSrv's callback endpoint.
- AuthenticateExternalAsync is called eventually with ExternalIdentity, local user is created and authorization code is returned to Client's backend (in browser). Backend-backend stuff goes on with tokens and user info, user is created in Client backend.
- Client's backend feeds custom URI scheme to browser to switch back to Client's native app and login as created user.

Basically this is the flow.
Does it look ok ? Need critical eye on this.
One thing I don't know how to do is the identity retrieval from authentication native app to IdSrv, how to do it best. I have seen facebook and google examples, but they have openid connect implemented themselves and in my case native application will not be an openid provider.

**Comment by brockallen** [comment]
I'm not following everything, which means there's confusion either on your side or this side.

> Need critical eye on this.

This sounds like you need formal security consulting.
